tests/io/vasp/test_inputs.py

```diff
@@ -721,51 +721,31 @@ def test_proc_types(self):
         assert Incar.proc_val("HELLO", "-0.85 0.85") == "-0.85 0.85"
 
     def test_check_params(self):
-        # Triggers warnings when running into nonsensical parameters
+        # Triggers warnings when running into invalid parameters
         with pytest.warns(BadIncarWarning) as record:
             incar = Incar(
                 {
                     "ADDGRID": True,
                     "ALGO": "Normal",
                     "AMIN": 0.01,
-                    "AMIX": 0.2,
-                    "BMIX": 0.001,
-                    "EDIFF": 5 + 1j,  # EDIFF needs to be real
-                    "EDIFFG": -0.01,
-                    "ENCUT": 520,
-                    "IBRION": 2,
                     "ICHARG": 1,
-                    "ISIF": 9,
-                    "ISMEAR": 1,
-                    "ISPIN": 2,
-                    "LASPH": 5,  # Should be a bool
-                    "LORBIT": 11,
-                    "LREAL": "Auto",
-                    "LWAVE": False,
                     "MAGMOM": [1, 2, 4, 5],
-                    "METAGGA": "SCAM",  # spelling mistake
-                    "NELM": 200,
-                    "NPAR": 4,
-                    "NSW": 99,
-                    "PREC": "Accurate",
-                    "SIGMA": 0.2,
-                    "NBAND": 250,  # spelling mistake
-                    "PHON_TLIST": "is_a_str",  # this parameter should be a list
-                    "LATTICE_CONSTRAINTS": [
-                        True,
-                        False,
-                        "f",
-                    ],  # Should be a list of bools
-                    "M_CONSTR": [True, 1, "string"],  # Should be a list of real numbers
```

Comment:
Meanwhile I noticed the type checking for the values of "LATTICE_CONSTRAINTS" and "M_CONSTR" was not implemented in the `check_params` method. As there isn't explicit definition for the type of list elements in `incar_parameters.json` (such as List[bool]).

Reply:
I don't know that it's possible to do more careful type checking the way with the current framework (e.g., `LATTICE_CONSTRAINTS : Sequence[bool, bool, bool]`) but that's good to note.